packages/gatsby/src/datastore/common/query.ts

```diff
@@ -261,3 +261,35 @@ export function objectToDottedField(
   })
   return result
 }
+
+const comparatorSpecificity = [
+  DbComparator.EQ,
+  DbComparator.IN,
+  DbComparator.GTE,
+  DbComparator.LTE,
+  DbComparator.GT,
+  DbComparator.LT,
+  DbComparator.NIN,
+  DbComparator.NE,
+]
```

Comment:
Wouldn't it be easier to have map/obj to be able to lookup specifity?
```
const comparatorSpecificity = {
  [DbComparator.EQ]: 1,
  //...etc
}
```

so then checking which should have prio would be just `` comparatorSpecificity[a.comparator] ?? throw new Error(`wat`) `` or something along those lines?

Reply:
Addressed in https://github.com/gatsbyjs/gatsby/pull/32261/commits/b6a8c581b768cc49aa843216cccba1eee999859f

Reply:
Addressed in https://github.com/gatsbyjs/gatsby/pull/32261/commits/b6a8c581b768cc49aa843216cccba1eee999859f